### ISSUE TITLE
BIOMOD_EnsembleForecasting- em.by

### ISSUE THREAD

**Issue by NadiaVic** [biomodhub/biomod2#374]
I have a question related to the way in which I have to assemble the models to be combined. 
I used in my biomod model 3 algorithms with three different PA datasets and repetitions.  
Initially, in the em.by parameter, I selected "all", but the biomod2 pdf says:
 "Be aware that all of these combinations are allowed, but some may not make
sense depending mainly on how pseudo-absence datasets have been built and whether all
of them have been used for all single models or not (see PA.nb.absences and models.pa
parameters in BIOMOD_FormatingData and BIOMOD_Modeling functions respectively)."

Thus, I was wondering, if it is better to assemble the models by em.by="PA + algo"?
I was searching any bibliography to understand how method choose, but I do not find it. 
Can you help me, please? 
Thanks in advance, 

```

**Comment by MayaGueguen** [comment]
Hello Nadia,

Indeed, the way you combine single models into ensemble depends on what are your modeling objectives :eyes: : 

- if you just want to have the "best" prediction possible for your species, we advise you to select `all` method
- if you would like to exlore variability in predictions between algorithms, select `algo`
- if you would like to exlore variability in predictions between for example selection of pseudo-absence points, select `PA`
- etc

Note that you can also use the `EMcv` ensemble model to see spatial areas where single models diverge in predictions.

You can also check the accuracy of your single models first with plot functions such as [bm_PlotEvalBoxplot](https://biomodhub.github.io/biomod2/reference/bm_PlotEvalBoxplot.html) and choose by yourself to remove set of repetitions of pseudo-absences, or algorithms, that seem to work less well compare to others.

In short, if your goal is to make predictions, just be sure to check that you have homogeneous evaluation scores between pseudo-absence sets (_meaning you sample PA correctly_) and between repetitions (_meaning you have enough data to calibrate your models_).

Hope it helps,
Maya
